From 3dd97e70450f899bad68f348e5be94c68a2f7b72 Mon Sep 17 00:00:00 2001
From: Jonathan Bell <jonathan@raspberrypi.com>
Date: Wed, 24 Jan 2024 16:32:13 +0000
Subject: [PATCH 0854/1085] dts: rp1: nobble HS and FS/LS park mode for dwc3
 xhci

Synopsys have recommended that we disable this feature entirely, as
Superspeed park mode can hang the controller, and Hs/FS/LS park mode can
cause performance degradation with bursty devices.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
---
 arch/arm/boot/dts/broadcom/rp1.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

--- a/arch/arm/boot/dts/broadcom/rp1.dtsi
+++ b/arch/arm/boot/dts/broadcom/rp1.dtsi
@@ -1018,6 +1018,8 @@
 			snps,axi-pipe-limit = /bits/ 8 <8>;
 			snps,dis_rxdet_inp3_quirk;
 			snps,parkmode-disable-ss-quirk;
+			snps,parkmode-disable-hs-quirk;
+			snps,parkmode-disable-fsls-quirk;
 			snps,tx-max-burst-prd = <8>;
 			snps,tx-thr-num-pkt-prd = <2>;
 			interrupts = <RP1_INT_USBHOST0_0 IRQ_TYPE_EDGE_RISING>;
@@ -1032,6 +1034,8 @@
 			snps,axi-pipe-limit = /bits/ 8 <8>;
 			snps,dis_rxdet_inp3_quirk;
 			snps,parkmode-disable-ss-quirk;
+			snps,parkmode-disable-hs-quirk;
+			snps,parkmode-disable-fsls-quirk;
 			snps,tx-max-burst-prd = <8>;
 			snps,tx-thr-num-pkt-prd = <2>;
 			interrupts = <RP1_INT_USBHOST1_0 IRQ_TYPE_EDGE_RISING>;
